mastering/conditional.rst

```diff
@@ -46,5 +46,56 @@ Here is an example of what we could do in our **configure method**:
           else:
              self.requires("OpenSSL/1.0.2d@lasote/stable")
 
+Constrain settings and options
+------------------------------
 
-.. seealso:: Check the section :ref:`Reference/conanfile.py/configure(), config_options() <method_configure_config_options>` to find out more.
+Sometimes there are libraries that are not compatible with specific settings like libraries
+that are not compatible with an architecture or options that only make sense for an operating system. It can be also useful when there are
+settings under development.
+
+There are two approaches for this situation:
+
+- **Use** ``configure()`` **to raise an error for non-supported configurations**:
+
+  This approach is the first one evaluated when Conan loads the recipe so it is quite handy to perform checks of the input settings. It
+  relies on the set of possible settings inside your *settings.yml* file so it can be used to constrain any recipe.
+
+  .. code-block:: python
+
+      def configure(self):
+          if self.settings.os == "Windows":
+            raise ConanException("This library is not compatible with Windows")
+
+  This same method is also valid for ``options`` and ``config_options()`` method and it is commonly used to remove options for one setting:
+
+  .. code-block:: python
+
+      def config_options(self):
+          if self.settings.os == "Windows":
+              del self.options.fPIC
+
+- **Constrain settings inside a recipe**:
```

Comment:
I am not a big fan of this feature. It is verbose and can become ugly if trying to constrain some subsetting. Generally prefer the configure() method and only for some special cases where it is 100% that such code will never, ever work with that setting. Please add some note or comment, suggesting this, if possible.